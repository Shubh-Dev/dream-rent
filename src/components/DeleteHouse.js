import { React, useState, useEffect } from 'react';
import axios from 'axios';
import './deletehouse.css';

const DeleteHouse = () => {
  const [isLoaded, setIsLoaded] = useState(false);
  const [houses, setHouses] = useState([]);

  useEffect(() => {
    axios.get('http://[::1]:3000/api/v1/houses')
      .then((res) => {
        setIsLoaded(true);
        setHouses(res.data);
      });
  }, []);

  const reRenderPage = (id) => {
    const newHouses = houses.filter((house) => house.id !== id);
    setHouses(newHouses);
  };

  if (!isLoaded) {
    return <div>Loading...</div>;
  }
  return (
    <div className="delete-house-container">
      { houses.map((house) => (
        <div className="card" key={house.id}>
          <img src={house.image_path} alt="house" className="delete-house-img" />
          <div className="card-body">
            <h5 className="card-title">
              Address:
              {house.address}
            </h5>
            <p className="card-text">
              House Type:
              {house.house_type}
            </p>
            <p className="card-text">
              <small className="text-muted">
                Rooms :
                {house.rooms}
              </small>
            </p>
            <p className="card-text">
              <small className="text-muted">
                Size:
                {house.size}
              </small>
            </p>
            <button type="button" className="delete-btn" onClick={() => { axios.delete(`http://[::1]:3000/api/v1/houses/${house.id}`); reRenderPage(house.id); }}>Delete</button>

          </div>

        </div>

      ))}

    </div>

  );
};

export default DeleteHouse;
