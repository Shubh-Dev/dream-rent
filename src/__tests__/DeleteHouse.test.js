import React from "react";
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import DeleteHouse from "../components/DeleteHouse";

describe('DeleteHouse', () => {
    it('renders correctly', () => {
        const tree = TestRenderer.create(<DeleteHouse />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders the DeleteHouse component', () => {
        render(<DeleteHouse />);
        expect(screen.getByText('DELETE HOUSE')).toBeInTheDocument();
    });
    }
);