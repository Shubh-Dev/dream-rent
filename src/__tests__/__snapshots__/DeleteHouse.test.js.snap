// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteHouse renders correctly 1`] = `
<div>
  Loading...
</div>
`;
